cardano-api/src/Cardano/Api/LedgerState.hs

```diff
@@ -0,0 +1,913 @@
+            }
+
+        clientIdle_DoneN
+          :: Nat n
+          -> IO (ClientPipelinedStIdle n (BlockInMode CardanoMode) ChainPoint ChainTip IO ())
+        clientIdle_DoneN n = case n of
+          Succ predN -> return (CollectResponse Nothing (clientNext_DoneN predN)) -- Ignore remaining message responses
+          Zero -> return (SendMsgDone ())
+
+        clientNext_DoneN
+          :: Nat n
+          -> ClientStNext n (BlockInMode CardanoMode) ChainPoint ChainTip IO ()
+        clientNext_DoneN n =
+          ClientStNext {
+              recvMsgRollForward = \_ _ -> clientIdle_DoneN n
+            , recvMsgRollBackward = \_ _ -> clientIdle_DoneN n
+            }
+
+        fromChainTip :: ChainTip -> WithOrigin BlockNo
+        fromChainTip ct = case ct of
+          ChainTipAtGenesis -> Origin
+          ChainTip _ _ bno -> At bno
+
+type LedgerStateHistory = Seq (SlotNo, LedgerState, WithOrigin (BlockInMode CardanoMode))
+
+--------------------------------------------------------------------------------
+-- Everything below was copied/adapted from db-sync                           --
+--------------------------------------------------------------------------------
+
+genesisConfigToEnv ::
+  -- DbSyncNodeParams ->
+  GenesisConfig ->
+  Either DbSyncNodeError Env
+genesisConfigToEnv
+  -- enp
+  genCfg =
+    case genCfg of
+      GenesisCardano _ bCfg sCfg
+        | Cardano.Crypto.ProtocolMagic.unProtocolMagicId (Cardano.Chain.Genesis.configProtocolMagicId bCfg) /= Shelley.Spec.sgNetworkMagic (scConfig sCfg) ->
+            Left . NECardanoConfig $
+              mconcat
+                [ "ProtocolMagicId ", textShow (Cardano.Crypto.ProtocolMagic.unProtocolMagicId $ Cardano.Chain.Genesis.configProtocolMagicId bCfg)
+                , " /= ", textShow (Shelley.Spec.sgNetworkMagic $ scConfig sCfg)
+                ]
+        | Cardano.Chain.Genesis.gdStartTime (Cardano.Chain.Genesis.configGenesisData bCfg) /= Shelley.Spec.sgSystemStart (scConfig sCfg) ->
+            Left . NECardanoConfig $
+              mconcat
+                [ "SystemStart ", textShow (Cardano.Chain.Genesis.gdStartTime $ Cardano.Chain.Genesis.configGenesisData bCfg)
+                , " /= ", textShow (Shelley.Spec.sgSystemStart $ scConfig sCfg)
+                ]
+        | otherwise ->
+            let
+              topLevelConfig = Consensus.pInfoConfig (mkProtocolInfoCardano genCfg)
+            in
+            Right $ Env
+                  { envLedgerConfig = Consensus.topLevelConfigLedger topLevelConfig
+                  , envProtocolConfig = Consensus.topLevelConfigProtocol topLevelConfig
+                  }
+
+readDbSyncNodeConfig :: NodeConfigFile -> IO DbSyncNodeConfig
+readDbSyncNodeConfig (NodeConfigFile ncf) = do
+    ncfg <- parseNodeConfig <$> readByteString ncf "node"
+    coalesceConfig ncfg (mkAdjustPath ncf)
+
+data NodeConfig = NodeConfig
```

Comment:
There is `NodeConfiguration` in `cardano-node` that has this information and can be extended if there are fields missing.

Reply:
I think I misunderstood this yesterday. Seeing as I'll be able to remove this once the query for ledger config gets merged, Perhaps we're ok with having `NodeConfig` for now? I'll add a TODO in the code.

Reply:
No worries. I think the best solution would be to move the node's configuration stuff to the api. This is good for now.

Reply:
Update: The genesis file will eventually not be needed so we don't need to move the node configuration stuff to cardano-api